src/Metadata/Resource/ResourceNameCollection.php

```diff
@@ -19,13 +19,15 @@
 final class ResourceNameCollection implements \IteratorAggregate, \Countable
 {
     private $classes;
+    private $context;
 
     /**
      * @param string[] $classes
      */
-    public function __construct(array $classes = [])
+    public function __construct(array $classes = [], array $context = [])
```

Comment:
Uhh... Thanks for the clarification. Perhaps we can cache the combined result of YAML parsing (be they from a single file, or one file for each resource), indexed by the resource class. So both YamlResourceNameCollectionFactory and YamlResourceMetadataFactory would have that same cache injected into them.